packages/valory/skills/market_manager_abci/bets.py

```diff
@@ -237,8 +253,10 @@ class BetsEncoder(json.JSONEncoder):
 
     def default(self, o: Any) -> Any:
         """The default encoder."""
-        if dataclasses.is_dataclass(o):
+        if dataclasses.is_dataclass(o) and not isinstance(o, type):
```

Comment:
Can you explain this?

Reply:
> Pyright: Argument of type "DataclassInstance | type[DataclassInstance]" cannot be assigned to parameter "obj" of type "DataclassInstance" in function "asdict"
>   Type "DataclassInstance | type[DataclassInstance]" is not assignable to type "DataclassInstance"
>     "__dataclass_fields__" is defined as a ClassVar in protocol

@Adamantios  I was getting above pyright issues, basically is_dataclass check would hold true for both dataclass instance and type[DataclassInstance], we wouldn't have faced any issue because during loading we use a bets file, but the check condition was not correct.